flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/exceptions/BigQueryConnectorException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.sink.exceptions;
+
+/** This class wraps errors found while connecting to BigQuery storage. */
+public class BigQueryConnectorException extends RuntimeException {
```

Comment:
i'm sure there's a reason but why wrap these exceptions in a new type if we arent enriching them?

Reply:
These wrapper exceptions will be enriched (similar to [spark conector](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/master/bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryConnectorException.java)) when more complex execution states are introduced with exactly-once.

Same reasoning applies to the serialization exception. That class will be enriched in upcoming iterations, post MVP.